app/controllers/dishes_controller.rb

```diff
@@ -0,0 +1,18 @@
+class DishesController < ApplicationController
+
+  def show
+    @dish = Dish.find(params[:id])
+  end
+
+  def update
```

Comment:
I would've pulled this functionality out into a separate joins controller instead of trying to do it all in a dishes#update action. For both of these cases, we are doing CRUD operations on the joins resource - DishIngredient. So you could make a DishIngredientsController and use a `create` and `destroy` actions to add an association and delete an association, respectively. This is a more RESTful implementation as you are not updating teh attributes of a dish resource, but rather creating and deleting the relationship between these 2 resources.